storage/bootstrap/result.go

```diff
@@ -22,14 +22,57 @@ package bootstrap
 
 import (
 	"github.com/m3db/m3db/storage/block"
+	"github.com/m3db/m3x/time"
 )
 
+type result struct {
+	results     ShardResults
+	unfulfilled ShardTimeRanges
+}
+
+// NewResult creates a new result.
+func NewResult() Result {
+	return &result{
+		results:     make(ShardResults),
+		unfulfilled: make(ShardTimeRanges),
+	}
+}
+
+func (r *result) ShardResults() ShardResults {
+	return r.results
+}
+
+func (r *result) Unfulfilled() ShardTimeRanges {
+	return r.unfulfilled
+}
+
+func (r *result) AddShardResult(shard uint32, result ShardResult, unfulfilled xtime.Ranges) {
+	if result != nil && len(result.AllSeries()) > 0 {
+		r.results[shard] = result
```

Comment:
Yeah I thought this was going to be handled elsewhere but on review seems it's not.  Might be best to merge with ShardResult AddResult(other ShardResult) method